connect/src/routes/cctp/automatic.ts

```diff
@@ -179,9 +179,21 @@ export class AutomaticCCTPRoute<N extends Network>
     let nativeGasAmount = 0n;
 
     if (nativeGasPerc > 0.0) {
+      const dcb = await request.toChain.getAutomaticCircleBridge();
+      let maxSwapAmount = await dcb.maxSwapAmount();
+      if (redeemableAmount < maxSwapAmount) {
+        // can't swap more than the receivable amount
+        maxSwapAmount = redeemableAmount;
+      }
+
       const scale = 10000;
-      const scaledGas = BigInt(nativeGasPerc * scale);
-      nativeGasAmount = (transferableAmount * scaledGas) / BigInt(scale);
+      const scaledGas = BigInt(Math.floor(nativeGasPerc * scale));
+      // the native gas percentage is applied to the max swap amount
+      nativeGasAmount = (maxSwapAmount * scaledGas) / BigInt(scale);
+      if (nativeGasAmount === redeemableAmount && nativeGasAmount > 0n) {
+        // edge case: transfer will revert if the native gas amount is equal to the redeemable amount
+        nativeGasAmount -= 1n;
```

Comment:
What is the significance of subtracting 1 bigint? Just to make it below zero?

Reply:
If the USDC amount we're swapping for native gas is equal to the amount we're transferring minus the relayer fee, then the contract will revert, so we need to make sure it's less than that in the rare case that it's equal